recipes/qpdf/all/conanfile.py

```diff
@@ -0,0 +1,165 @@
+        "with_crypto": ["native", "openssl", "gnutls"],
+        "with_jpeg": ["libjpeg", "libjpeg-turbo", "mozjpeg"],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_crypto": "native",
+        "with_jpeg": "libjpeg",
+    }
+
+    @property
+    def _minimum_cpp_standard(self):
+        return 14
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "gcc": "5",
+            "clang": "3.4",
+            "apple-clang": "10",
+        }
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            try:
+                del self.options.fPIC
+            except Exception:
+                pass
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        # https://qpdf.readthedocs.io/en/stable/installation.html#basic-dependencies
+        self.requires("zlib/1.2.12")
+        if self.options.with_crypto == "openssl":
+            self.requires("openssl/1.1.1q")
+        elif self.options.with_crypto == "gnutls":
+            raise ConanInvalidConfiguration("GnuTLS is not available in Conan Center yet.")
+        if self.options.with_jpeg == "libjpeg":
+            self.requires("libjpeg/9e")
+        elif self.options.with_jpeg == "libjpeg-turbo":
+            self.requires("libjpeg-turbo/2.1.4")
+        elif self.options.with_jpeg == "mozjpeg":
+            self.requires("mozjpeg/4.0.0")
+
+
+    def validate(self):
+        if self.info.settings.compiler.cppstd:
+            check_min_cppstd(self, self._minimum_cpp_standard)
+        if is_msvc(self):
+            check_min_vs(self, "150")
+        else:
+            minimum_version = self._compilers_minimum_version.get(str(self.info.settings.compiler), False)
+            if minimum_version and Version(self.info.settings.compiler.version) < minimum_version:
+                raise ConanInvalidConfiguration(f"{self.ref} requires C++{self._minimum_cpp_standard}, which your compiler does not support.")
+        if self.options.shared == True and str(self.settings.os) in ["Macos"]:
+            raise ConanInvalidConfiguration("shared build currently produces linker errors on mac build systems")
+
+    def build_requirements(self):
+        self.tool_requires("cmake/3.24.1")
+        self.tool_requires("pkgconf/1.9.3")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version],
+                  destination=self.source_folder, strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["BUILD_STATIC_LIBS"] = not self.options.shared
+        # https://qpdf.readthedocs.io/en/latest/installation.html#build-time-crypto-selection
+        tc.variables["USE_IMPLICIT_CRYPTO"] = False
```

Comment:
Try defining `tc.cache_variables["CMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP"] = True` here to fix your Windows error.

```suggestion
tc.variables["USE_IMPLICIT_CRYPTO"] = False
tc.cache_variables["CMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP"] = True
```

Reply:
I added in hope to fix. It seems that the problem may be the same for mac. When this come green through, i will try to re-enable the mac shared builds once more.

Reply:
This will fix an issue specific to your Windows builds failing. I saw your fix for macOS shared libraries. Since they are using an `OBJECT` library, declaring `PRIVATE` linkage won't actually propagate to wherever the library is actually linked. So, your fix looks correct there. I hope it works.